src/plugins/mf_priority.cpp

```diff
@@ -1129,10 +1133,69 @@ static int job_updated (flux_plugin_t *p,
         bank_it = lookup_result.second;
     }
 
-    // fetch the priority of the validated queue; assign it to the bank_info
-    // struct associated with the job
-    int queue_factor = get_queue_info (queue, bank_it);
-    b->queue_factor = queue_factor;
+    // if the queue for the job has been updated, fetch the priority of the
+    // validated queue and assign it to the associated bank_info struct
+    if (queue != NULL) {
+        int queue_factor = get_queue_info (queue, bank_it);
+        b->queue_factor = queue_factor;
+    }
+
+    return 0;
+}
+
+
+/*
+ *  check for an updated queue and validate it for a user/bank; if the
+ *  user/bank does not have access to the queue they are trying to update
+ *  their job for, reject the update and keep the job in its current queue.
+ */
+static int update_queue_cb (flux_plugin_t *p,
+                            const char *topic,
+                            flux_plugin_arg_t *args,
+                            void *data)
+{
+    std::map<std::string, struct bank_info>::iterator bank_it;
+    int userid;
+    char *bank = NULL;
+    char *queue = NULL;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:s, s?s}",
+                                "userid", &userid,
+                                "value", &queue,
+                                "attributes.system.bank", &bank) < 0)
```

Comment:
Oops! Sorry about that. Yes, I do mean to be unpacking the `bank` from the jobspec - I think I thought I could shorthand the traditional way of unpacking it from jobspec, just like you suggested above. Since the tests were passing, I figured this was okay too...

Perhaps the reason it passes with this is because in this set of tests, the user is submitting their job under a default bank, and thus `flux_jobtap_jobspec_update_pack ()` is being called to add the bank to the main eventlog?

Either way, I can fix this to unpack it like how I always have.

Reply:
Maybe also look into adding a test that exercises this code when the user sets a non-default bank? Just in case :-)

Reply:
sure thing! i can add another set of tests to `t1030` that basically repeats the same workflow, but under a default bank. i'll push up a commit to this branch if that's okay.

Reply:
btw, i was curious, so i went back and tried submitting a job under a non-default bank and using what i had originally proposed in the callback for `job.update.attributes.system.queue`:

```
                                "{s:i, s:s, s?s}",
                                "userid", &userid,
                                "value", &queue,
                                "attributes.system.bank", &bank) < 0)
```

and the tests failed. So thanks for catching that!